controllers/k8ssandra/telemetry_reconciler.go

```diff
@@ -0,0 +1,86 @@
+// Logic in this file reconciles telemetry resources into the state declared in the CRs.
+
+package k8ssandra
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	telemetryapi "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/telemetry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *K8ssandraClusterReconciler) reconcileCassandraDCTelemetry(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	dcTemplate api.CassandraDatacenterTemplate,
+	actualDc *cassdcapi.CassandraDatacenter,
+	logger logr.Logger,
+	remoteClient client.Client,
+) (ctrl.Result, error) {
+	logger.Info("reconciling telemetry", "cluster", kc.Name, "datacenter", actualDc.Name)
+	mergedSpec := kc.Spec.Cassandra.CassandraTelemetry.Merge(dcTemplate.CassandraTelemetry)
```

Comment:
> We are only deploying ServiceMonitors, which is basically just configuration. Why do we this in the status at all?

The reason I think status may be important is that any telemetry we integrate with runs the risk that the dependancies (e.g. Prometheus operator) are not installed. In my view, things are very likely to go wrong at this edge of the system, because we have more external dependancy and less control over the things we are integrating with.

The logical mitigations are validating webhooks and clear status/problem reporting to the user.

Those considerations notwithstanding, I am very open to removing all status-related code out of this PR, at the very least. Setting status is actually quite complex (not least due to the problems @adutra was raising where you have to change the way that e.g. Stargate and Reaper set status whenever you add an additional status field).

Let me know which way you want to go.

Reply:
> To be consistent, this should be done in the Coalesce function in datacenter.go.

I'm not sure I understand your intention here. Coalesce merge a `*api.CassandraClusterTemplate` with a `*api.CassandraDatacenterTemplate` to produce a `DatacenterConfig`. 

I'm merging two `TelemetrySpec` objects to produce a `TelemetrySpec` object, and that's all my reconciler needs to proceed, it doesn't need to know about the entire DC, and `Coalesce` doesn't seem to do what I need it to?